src/js/components/conversation.jsx

```diff
@@ -62,7 +62,32 @@ export class ConversationComponent extends Component {
     }
 
     render() {
-        const messages = this.props.conversation.messages.map((message) => {
+        const processedMessages = [...this.props.conversation.messages];
+        let lastAuthor;
+        processedMessages.forEach((message, index) => {
+            const author = message.role === 'appUser' ? message.role : `${message.role}_${message.name}`;
```

Comment:
This naming convention (`${message.role}_${message.name}`) would mean that whispers and replies from the same person will not be grouped, right? Maybe the prefix should just be `am_` or something